### PR TITLE
enable configurable memory size in DTS generation; bump NR_CPUS to 32 in linux_defconfig

### PR DESCRIPTION
**Pull request by muwyse**
This PR makes the following changes to our linux build and infrastructure:

1. enable configurable memory size for linux in the DTS generation. BP supports up to 2 GiB of cacheable DRAM currently. FPGA boards with larger memory chips can use the full cacheable DRAM space. Default size remains 64 MiB for testing and small memory systems.
2. tweaks the DTS generation to preserve copies of the .dts and .dtb for reference after building
3. removes lingering reference to non-existent cfg/linux.patch
4. bump the CONFIG_NR_CPUS to 32 in linux_defconfig, which defines the maximum number of CPUs. This may increase the linux kernel image size slightly. 32 is the max supported for risc-v isa in the kernel.

These changes have been tested and confirmed working on a vcu128 FPGA design, booting 8 cores in Linux and showing 2 GiB of memory usable by linux (`cat /proc/meminfo`) as well as a very simple smoke test on same 8-core FPGA design that can read and write the last 64-bit dword of the 2 GiB cacheable dram address space.

Verification on zynq-parrot may be a good second smoke test to ensure nothing breaks there.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -70,9 +73,14 @@ $(vmlinux_stripped): $(vmlinux)
...
 	$(LINUX_TARGET)-objcopy -O binary $< $@
 
-$(fw_payload): $(opensbi_srcdir) $(vmlinux_binary)
+$(bp_dts):
 	mkdir -p $(opensbi_wrkdir)/platform/blackparrot
-	python $(GENDTS_PY) --ncpus=$(OPENSBI_NCPUS) | dtc -O dtb -o $(opensbi_wrkdir)/platform/blackparrot/blackparrot.dtb
+	python $(GENDTS_PY) --ncpus=$(OPENSBI_NCPUS) --mem-size=$(MEM_SIZE) > $(bp_dts)
+
+$(bp_dtb): $(bp_dts)
+	dtc -O dtb -o $(bp_dtb) $<
```
Comment (dpetrisko) [inline Makefile]: Can you do DTB ?= dtb in this file?

File: cfg/test_info.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+echo "Running test.sh"
```
Comment (dpetrisko) [inline cfg/test_info.sh:3]: Can we run this by default? Seems more useful than just shutting down

File: Makefile (hunk 1 above)
Comment (muwyse) [reply Makefile]: If this is a request to add `DTB ?= ...`, yes that can be done if there is a desire to be able to override the gendts.py produced device tree.

If just a question, there is not currently a variable to do that override. The .dts file is generated by gendts.py, which is then fed to dtc to create the binary version (.dtb). Arguments to gendts.py allow us to set number of processors and memory size in the device tree.

File: cfg/test_info.sh (hunk 2 above)
Comment (muwyse) [reply cfg/test_info.sh:3]: That would be my preference, but I was erring on the side of don't touch the defaults, hence the new script. I can make the existing cfg/test.sh look like this, or we can keep both and I can just point to test_info.sh by default. Preference?

Comment (dpetrisko) [reply cfg/test_info.sh:3]: My preference is to have test_shutdown.sh and test_info.sh and point to test_info by default

File: Makefile (hunk 1 above)
Comment (dpetrisko) [reply Makefile]: Sorry typo... DTC ?= dtc

**Review by dpetrisko — APPROVED** [review]
Nice!

File: Makefile (hunk 1 above)
Comment (muwyse) [reply Makefile]: ah, lol, got it. I will add that.

File: cfg/test_info.sh (hunk 2 above)
Comment (muwyse) [reply cfg/test_info.sh:3]: I will rename test.sh to test_shutdown.sh here. Pointing to test_info by default will happen in the SDK PR.

File: Makefile (hunk 1 above)
Comment (dpetrisko) [reply Makefile]: Similarly PYTHON ?= python

**Pull request merged**